src/libsync/connectionvalidator.cpp

```diff
@@ -143,22 +145,21 @@ void ConnectionValidator::slotStatusFound(const QUrl &url, const QJsonObject &in
     }
 
     // now check the authentication
-    if (_account->credentials()->ready())
-        QTimer::singleShot(0, this, SLOT(checkAuthentication()));
-    else
-        reportResult(CredentialsMissingOrWrong);
+    QTimer::singleShot( 0, this, SLOT( checkAuthentication() ));
 }
 
 // status.php could not be loaded (network or server issue!).
 void ConnectionValidator::slotNoStatusFound(QNetworkReply *reply)
 {
     auto job = qobject_cast<CheckServerJob *>(sender());
     qCWarning(lcConnectionValidator) << reply->error() << job->errorString() << reply->peek(1024);
-    if (!_account->credentials()->ready()) {
-        // This could be needed for SSL client certificates
-        // We need to load them from keychain and try
-        reportResult(CredentialsMissingOrWrong);
-    } else if (!_account->credentials()->stillValid(reply)) {
+    if (reply->error() == QNetworkReply::SslHandshakeFailedError) {
```

Comment:
I like the fact that this is more accurate than reporting wrong credentials for every error.